src/lib/mina_lib/mina_lib.ml

```diff
@@ -334,15 +334,47 @@ let active_or_bootstrapping =
         ~f:(Fn.const (Some ())) )
 
 let create_sync_status_observer ~logger ~is_seed ~demo_mode
-    ~transition_frontier_and_catchup_signal_incr ~online_status_incr
-    ~first_connection_incr ~first_message_incr =
+    ~(constraint_constants : Genesis_constants.Constraint_constants.t)
+    ~consensus_constants ~transition_frontier_and_catchup_signal_incr
+    ~online_status_incr ~first_connection_incr ~first_message_incr =
   let open Mina_incremental.Status in
+  let pre_genesis =
+    let fork_start_time =
+      let constants = consensus_constants in
+      Option.map constraint_constants.fork ~f:(fun x ->
+          Consensus.Global_slot.(
+            start_time ~constants
+              (of_slot_number x.previous_global_slot ~constants))
+          |> Block_time.to_time )
+    in
+    let genesis_time =
+      Block_time.to_time consensus_constants.genesis_state_timestamp
+    in
+    let start_time =
+      match fork_start_time with
+      | None ->
+          genesis_time
+      | Some fork_time ->
+          Time.max genesis_time fork_time
```

Comment:
I'm not sure if I understand the use fork_start_time here. It is always going to be some time in the future because we are calculating it using the new genesis timestamp set for the fork. So a node will show the wait status till it reaches that slot whereas we want it only till the genesis time.

Reply:
Ah ok, so genesis timestamp gets reset when we do a fork? Wasn't sure if it was the original genesis time or the fork one. So I guess in practice it will be the same as fork_time more or less right?

Reply:
Yeah the genesis timestamp gets reset for hard-forks

Reply:
fixed